### PR TITLE
Site Editor Sidebar: Add "Areas" details panel to all templates and update icon

### LINKED ISSUE #54933: Add template part shortcuts to all template detail panels
For some templates (e.g. Index) the detail panel includes shortcuts to view any of the template parts within that template (note the "Areas" section at the bottom):

In the interest of consistency it might be useful to add this feature to the detail panels for other templates too.

### PR DESCRIPTION
**Pull request by t-hamano**
Closes #54933

## What?

This PR adds "Areas" details panel to all templates.

symbolfilled

## Why?

The "Areas" details panel displays a list of template parts that the current template contains and allows you to quickly access those template parts.

In the current implementation, this panel is included in the `HomeTemplateDetails` component, so it [only appears if the template's `slug` is `home` or `index`](https://github.com/WordPress/gutenberg/blob/8f3392b35b0503dfd7d05768db065390b11c3391/packages/edit-site/src/components/sidebar-navigation-screen-template/index.js#L46-L49).

However, this panel should not be limited to template types.

## How?

I took the code for the Areas details panel contained in `HomeTemplateDetails` and split it out as a `TemplateAreas` component. It then renders this component on the sidebar screen of all templates.

_**Update**_: Additionally, I used the `symbolFilled` icon for general template parts so it matches the one you find in the block toolbar / inspector.

## Testing Instructions

- Go to the templates page in the site editor.
- The template parts that the template contains should be displayed in the Areas details panel.
- For templates that do not have template parts, such as TT2's Blank template, the Areas details panel should not be displayed.

### CODE REVIEW

File: packages/edit-site/src/components/sidebar-navigation-screen-template/template-areas.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+						),
+						...templatePart,
+						clientId: block.clientId,
+					} ) )
+			: [];
+	}, [ currentTemplateParts, templatePartAreas ] );
+
+	if ( ! templateAreas.length ) {
+		return null;
+	}
```
Comment (t-hamano) [inline packages/edit-site/src/components/sidebar-navigation-screen-template/template-areas.js:111]: While working on this PR, I encountered a bug where only the title "Areas" was rendered even when the template had no template parts. This code is to fix that.

**Comment by ramonjd** [comment]
This PR looks fine to me 👍🏻 I'll wait for @jameskoster to take it for a test run. 

It does bring more attention to the unexpected state of the back button, whereby clicking back from a template area won't return you to the template. This is the current state in trunk however.

**Comment by jameskoster** [comment]
Thanks for the PR, it's working well for me :) 

My only suggestion would be to use the `symbolFilled` icon for general template parts so it matches the one you find in the block toolbar / inspector.

**Comment by t-hamano** [comment]
Thanks for the review, @ramonjd, @jameskoster!

> My only suggestion would be to use the `symbolFilled` icon for general template parts so it matches the one you find in the block toolbar / inspector.

Makes sense, I made such a change in 998a75f. Furthermore, there was a lot of unnecessary code left, so I processed it with [5855481](https://github.com/WordPress/gutenberg/pull/55677/commits/58554812423ecec18ad58eb340f43960f9223f45).

symbolfilled

**Review by jameskoster — APPROVED** [review]
In terms of design this is looking good. Thanks :)

**Comment by t-hamano** [comment]
Sorry for the late reply, I forgot to ship this PR 😅 I updated this PR again with the latest trunk and it still works as intended, so I'd like to merge it as soon as the tests pass.

**Pull request merged**